api/features/serializers.py

```diff
@@ -247,6 +251,11 @@ def validate(self, attrs):
 
         return attrs
 
+    def get_feature_state(self, instance: Feature) -> FeatureState | None:
+        for feature_state in self.context.get("feature_states", []):
```

Comment:
Couldn't we build a dictionary when we load it into the context, then each lookup here is O(1) rather than O(n)?

Reply:
Since we're dealing with small collections of lists I don't think it really impacts performance, but I've re-written it to using dictionaries because it was easy to do so.

Reply:
Well, some projects will have upwards of 200 features, I think at that scale it could make a difference, no? If we're doing O(200), 200 times.

Reply:
I was under the impression that the page length was 20 records. If we're dealing with O(200) 200 times it does start to add up.